### PR TITLE
fix can't config passport#initialize in middleware.json

### PR DESCRIPTION
**Pull request by clarkorz**
support passport#initialize and passport#authenticate

Fix strongloop/loopback#954

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by bajtos** [comment]
Hi @clarkorz, thank you for the patch, it looks mostly good. Please rebase it on top of the current master, move `test/helpers/passport.js` to `test/fixtures` and squash all commits into a single one.

File: lib/executor.js [hunk 1]
```diff
@@ -274,7 +274,10 @@ function setupMiddleware(app, instructions) {
         data.fragment ? ('#' + data.fragment) : '');
     var factory = require(data.sourceFile);
     if (data.fragment) {
-      factory = factory[data.fragment];
+      if (typeof factory == 'function')
+        factory = factory[data.fragment];
+      else
+        factory = factory[data.fragment].bind(factory);
```
Comment (bajtos) [inline lib/executor.js]: What is the point of the first if-branch? Why can't we bind the factory method always?

**Comment by clarkorz** [comment]
@bajtos I have created a new PR here https://github.com/strongloop/loopback-boot/pull/92

**Comment by bajtos** [comment]
Please don't open a new pull request next time, learn how to rewrite git history instead: `git commit --amend`, `git rebase -i master`, `git cherry-pick` and of course `git push -f`.

**Pull request closed without merging**